bubble_sort_by.rb

```diff
@@ -0,0 +1,15 @@
+def bubble_sort_by(arr)
+  counter = 1 # initialized to 1.
+  while counter <= arr.length - 1 # Check if we have run length-1 times the code
+    arr.each_with_index do |_value, index| # using this method to have a self updated index done by the method
+      next unless index < arr.length - 1 # as long as the index is less than the length of the array
+
+      arr[index], arr[index + 1] = arr[index + 1], arr[index] if yield(arr[index], arr[index + 1]).positive?
+      # if the yield(block) evaluates to a number > than 0 : execute
```

Comment:
* [ ] comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
make sure this rule is observed in other projects as it may lead to the project not to be approved

Reply:
thanks for the feedback . Will took them into consideration. Carlos